scripts/gitOpsApi.js

```diff
@@ -0,0 +1,213 @@
+         */
+        const getTree = async function (repoOwner, repoName, branch) {
+            return getBranchRef(repoOwner, repoName, branch).then((commit) => {
+                parentCommit = commit;
+                return octokit.request("GET /repos/{owner}/{repo}/git/trees/{tree_sha}", {
+                    owner: repoOwner,
+                    repo: repoName,
+                    tree_sha: commit.data.object.sha
+                });
+            });
+        };
+
+        return Promise.all(files.map((file) => {
+            return octokit.request("POST /repos/{owner}/{repo}/git/blobs", {
+                owner: repoOwner,
+                repo: repoName,
+                content: file.content
+            });
+        })).then((blobs) => {
+            return getTree(repoOwner, repoName, branch).then(function (tree) {
+                return octokit.request("POST /repos/{owner}/{repo}/git/trees", {
+                    owner: repoOwner,
+                    repo: repoName,
+                    tree: files.map(function(file, index) {
+                        return {
+                            path: file.path,
+                            mode: '100644',
+                            type: 'blob',
+                            sha: blobs[index].data.sha
+                        };
+                    }),
+                    base_tree: tree.data.sha
+                });
+            });
+        }).then((tree) => {
+            return octokit.request("POST /repos/{owner}/{repo}/git/commits", {
+                owner: repoOwner,
+                repo: repoName,
+                message: commitMsg,
+                tree: tree.data.sha,
+                parents: [parentCommit.data.object.sha]
+            });
+        }).then((commit) => {
+            return octokit.request("PATCH /repos/{owner}/{repo}/git/refs/{ref}", {
+                owner: repoOwner,
+                repo: repoName,
+                ref: "heads/" + branch,
+                sha: commit.data.sha
+            });
+        });
+    },
+
+	/**
+	 * Issue pull request against the data repo on behalf of an issuer account
+	 * @param {String} githubGraphqlAPI - The URL of the Github GraphQL API
+	 * @param {String} issuerGithubId - The Github account that the pull request will be issued on behalf of
+	 * @param {String} issuerAccessToken - The personal access token of the Github account that the pull request will
+	 * be issued on behalf of
+	 * @param {String} branchName - The name of the remote branch
+	 * @param {String} publishedDate - The published date of the new data files. Used in the commit message
+	 * @return {String|Object} return the URL of the issued pull request when the pull request is issued successfully.
+	 * otherwise, return the error object in a structure of: {isError: true, message: detailed-error-message}
+	 */
+	issuePullRequest: async (githubGraphqlAPI, issuerGithubId, issuerAccessToken, branchName, publishedDate) => {
```

Comment:
This API might be more usable if it didn't have side-effects by reading the environment variables - these should be read from function arguments, as well as the "octokit" instance as well. It might be an idea and more readable to reorganise these arguments as an "options" structure rather than accumulating more and more function arguments